lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/classify/stats/CubePercentile.java

```diff
@@ -0,0 +1,60 @@
+package edu.stanford.futuredata.macrobase.analysis.classify.stats;
+
+import com.google.common.collect.HashMultiset;
+import com.google.common.collect.Multiset;
+
+import java.util.*;
+
+/**
+ * Computes percentiles when given an array of metrics and the count of the
+ * number of times each occurs. Useful for computing percentiles on cubed data.
+ */
+public class CubePercentile {
+    private double[] counts;
+    private double[] metrics;
+
+    // Computed
+    private int numRawMetrics = 0;
+    private Multiset<Double> metricCounts;
```

Comment:
I think Multiset doesn't actually store duplicate elements. It seems to me like a version of Map that's good at counting, which is exactly what I want here, so it doesn't store much more than an array.

I don't think arrays are good here, because we don't know the number of unique metrics ahead of time, and we can't efficiently find and update elements with counts. Lists also run into the problem of finding and updating. So we want a map-like data structure.

I coded up a version of WeightedMetrics[] and benchmarked it. It's much slower than the current implementation.